src/3p.js

```diff
@@ -154,6 +154,42 @@ export function checkData(data, allowedFields) {
 }
 
 /**
+ * Utility function to perform a potentially asynchronous task
+ * exactly once for all frames of a given type and the provide the respective
+ * value to all frames.
+ * @param {!Window} global Your window
+ * @param {string} taskId Must be not conflict with any other global variable
+ *     you use. Must be the same for all callers from all frames that want
+ *     the same result.
+ * @param {function(*)} cb Callback function that is called when the work is
+ *     done. The first argument is the result.
+ * @param {function(function(*))} work Function implementing the work that
+ *     is to be done. Receives a second function that should be called with
+ *     the result when the work is done.
+ */
+export function computeInMasterFrame(global, taskId, cb, work) {
+  if (global.context.isMaster) {
+    const tasks = global[taskId] ? global[taskId] : [];
```

Comment:
Should we be giving scripts access to the direct `global`? Maybe create a `global.tasks` object instead?